### PR TITLE
Update the quickstart examples for Azure to not specify the location explicitly

### LINKED ISSUE #1535: Update the Azure examples to not flow the location, as that's no longer required
Related to #1173.

### PR DESCRIPTION
**Pull request by praneetloke**
Fixes #1535. Also requires https://github.com/pulumi/templates/pull/60.

### CODE REVIEW

File: content/docs/reference/clouds/azure/_index.md [hunk 1]
```diff
@@ -58,6 +58,7 @@ The Azure provider is open source and available in the [pulumi/pulumi-azure](htt
 The Azure provider accepts the following configuration settings.  These can be provided to the default Azure provider via `pulumi config set azure:<option>`, or passed to the constructor of `new azure.Provider` to construct a specific instance of the Azure provider.
 
 * `environment`: (Required) The cloud environment to use. It can also be sourced from the ARM_ENVIRONMENT environment variable. Supported values are: `public` (default), `usgovernment`, `german`, `china`.
+* `location`: (Required) The location to use. ResourceGroups will consult this property for a default location, if one was not supplied explicitly.
```
Comment (praneetloke) [inline content/docs/reference/clouds/azure/_index.md]: I also went ahead and added this so users are aware of this feature. I think it is super useful for Azure users. I assume we don't auto-generate this page?

Comment (joeduffy) [reply content/docs/reference/clouds/azure/_index.md]: It is optional, right? (It says "(Required)".)

Comment (praneetloke) [inline content/docs/reference/clouds/azure/_index.md]:
```suggestion
* `location`: (Optional) The location to use. ResourceGroups will consult this property for a default location, if one was not supplied explicitly.
```

File: content/docs/quickstart/azure/review-project.md [hunk 2]
```diff
@@ -21,14 +21,11 @@ const pulumi = require("@pulumi/pulumi");
...
 // Create an Azure Resource Group
-const resourceGroup = new azure.core.ResourceGroup("resourceGroup", {
-    location: "WestUS",
-});
+const resourceGroup = new azure.core.ResourceGroup("resourceGroup");
 
 // Create an Azure resource (Storage Account)
 const account = new azure.storage.Account("storage", {
     resourceGroupName: resourceGroup.name,
-    location: resourceGroup.location,
```
[the remark this answers is not in the document]
Comment (praneetloke) [reply content/docs/quickstart/azure/review-project.md]: I purposefully left the comment out. I added a full explanation below the code block. I can add the comment in the code block too, to match-up with the template, if you still feel they should match.

[the remark this answers is not in the document]
Comment (justinvp) [reply content/docs/quickstart/azure/review-project.md]: I was aiming for matching with the template, so if users are following along, they'll see the same thing in both the docs and on their machine, but I don't feel too strongly about it if you think it's better not to include the comment here. I'm sure there will eventually be some drift anyway as we make edits to the templates and forget to make corresponding changes here.

[the remark this answers is not in the document]
Comment (praneetloke) [reply content/docs/quickstart/azure/review-project.md]: > I'm sure there will eventually be some drift anyway as we make edits to the templates and forget to make corresponding changes here.

Yes, this is what I am concerned about matching closely with the templates. I am actually wondering if we could embed code files here. I was thinking that we could create a separate tutorials repo and each step would be its own folder under the main tutorial folder. This way, users could clone the repo and follow along, as well as we could embed the relevant step's code file here. Obviously, not for this PR but as a future improvement perhaps? Just an idea.

**Pull request merged**